app/controllers/mailbox_controller.rb

```diff
@@ -2,43 +2,17 @@ class MailboxController < ApplicationController
   before_filter :authenticate_user!
 
   def index
-    if user_signed_in?
-      @messages = current_user.mailbox.inbox
-      @messages.each{|m| m.mark_as_read(current_user)}
-    else
-      @messages =[]
-    end 
+    @messages = UserMailbox.new(current_user).inbox
```

Comment:
:+1: to isolating this in a private method, so it only has to change in one place